### ISSUE TITLE
Interceptable Popover

### ISSUE THREAD

**Issue by ant45de** [ant-design/ant-design#9240]
### What problem does this feature solve?
I have a antd-popover in which my user can enter form data. In case he closes the popover by clicking somewhere outside the popover I want to make sure that there will be no data loss. In the current Implementation of the popover the process of closing can not be intercepted. It would be useful to make it interceptable.

### What does the proposed API look like?
Basically the PopoverComponent should accept a prop *confirmClose* which is a function in which the parent can return true/false wether he wants to execute the close or not.

**Comment by yesmeck** [comment]
You can control Popover with `visible` and `onVisibleChange`.
